llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -198,6 +198,17 @@ static void ComputePTXValueVTs(const TargetLowering &TLI, const DataLayout &DL,
     return;
   }
 
+  // Given an array type, recursively traverse the elements with custom ComputePTXValueVTs.
+  if (ArrayType *ATy = dyn_cast<ArrayType>(Ty)) {
+    Type *EltTy = ATy->getElementType();
+    uint64_t EltSize = DL.getTypeAllocSize(EltTy);
+    for (unsigned i = 0, e = ATy->getNumElements(); i != e; ++i) {
```

Comment:
Nit:  `for (auto I : llvm::seq<int>(ATy->getNumElements()))` may be a bit cleaner than the 'classic' for loop.

Reply:
Interesting! Thank you for nit! Done